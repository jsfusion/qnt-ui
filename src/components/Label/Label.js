import React from 'react';
import { PropTypes } from 'prop-types';
import { ThemeProvider } from 'styled-components';
import { themeProps } from '../../settings';
import { StyledLabel } from './styles/Label.styles';

const { classPrefix } = themeProps;

const fontWeights = {
  100: '100',
  200: '200',
  300: '300',
  400: '400',
  500: '500',
  600: '600',
  700: '700',
  800: '800',
  900: '900',
  bold: 'bold',
  bolder: 'bolder',
};

const Label = React.forwardRef(
  (
    {
      children,
      id,
      fontWeight = '100',
      isItalic = false,
      ...rest
    },
    ref
  ) => {
    return (
      <ThemeProvider theme={{}}>
          <StyledLabel
            id={id}
            ref={ref}
            className={`${classPrefix}-label`}
            fontWeight={fontWeight}
            isItalic={isItalic}
            {...rest}
          >
            {children}
          </StyledLabel>
      </ThemeProvider>
    );
  },
);

Label.propTypes = {
  id: PropTypes.string,
  fontWeight: PropTypes.oneOf(Object.keys(fontWeights)),
  isItalic: PropTypes.bool,
}

export default Label;
