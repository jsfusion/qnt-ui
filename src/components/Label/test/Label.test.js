import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Label from '..';

configure({ adapter: new Adapter() });

let wrapper;
const props = {
  id: 'test_Label_id',
  fontWeight: 'bold',
  isItalic: true,
};

describe('***--- Label Component ---***', () => {
  beforeAll(() => {
    wrapper = shallow(<Label {...props}>Label text</Label>);
  });

  it('To match snapshot of Label component', async (done) => {
    expect(wrapper).toMatchSnapshot();
    done();
  });
});