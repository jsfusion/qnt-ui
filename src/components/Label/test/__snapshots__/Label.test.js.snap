// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`***--- Label Component ---*** To match snapshot of Label component 1`] = `ShallowWrapper {}`;
